### ISSUE TITLE
Taylor Diagram errors

### ISSUE THREAD

**Issue by kaabre** [ACME-Climate/acme_diags#128]
I'm encountering a few weird errors running the new version of acme_diags on Edison.  When running from the command line, I get a display variable error (that Susannah suggested may go away when X windows are used; this is indeed the case when I run on nomachine).  I did not have this error before the update.  However, even on nomachine where I don't get the display variable error, I get empty Taylor plots, and the csvs that should populate them are also empty.

The parameter file used and the outputs are located here: https://portal.nersc.gov/project/PNNL-PJR/bren937/scratch/TEST/newvers/ 

The logfiles are within each respective directory: one run on nomachine, one run on command line.  (Even though both were created via batch scripts to the queue using the same parameters)

(I get the same issue running on Constance, although I know that's a machine you don't support.  On acme2, I don't know what's going on but the acme_diags environment won't seem to update following the instructions you gave in the release notes.)

Are there any steps I am missing getting it to work on Edison/do you have expected output on that machine?

Thanks!

**Comment by zshaheen** [comment]
@kaabre Regarding the display variable error, we just released a new version, v1.2.1, that fixes this. It was because we had an `import matplotlib` statement without `matplotlib.use('Agg')` following it.

If you're using the non-dev env, using just `conda upgrade acme_diags` should get you the new version. Otherwise if on a dev env, just pull the latest code from master, install it, and it should be good.

**Comment by chengzhuzhang** [comment]
Notes for @zshaheen  and myself
For a model versus model run, all the json files that pass into the csv tables are there, but csv for tables are not generated..

Taylor diags are not supported for model versus model config right now.

**Comment by zshaheen** [comment]
@kaabre Thanks for finding this bug! It seems like Taylor diagrams don't work for model vs model comparisons, for now.

**Comment by kaabre** [comment]
Thanks!  Good to know -- I saw the related update -- I'm trying to run it now on a model-observation comparison and see what happens.

**Comment by kaabre** [comment]
I DO still run into the "invalid DISPLAY variable" when using the command line (rather than through nomachine).

**Comment by zshaheen** [comment]
@kaabre Can you confirm that you're on `v1.2.1`? Please try the command `python -c 'import acme_diags; print(acme_diags.__version__)'`. It should print out `v1.2.1`
```
(acme_diags_env_old) shaheen2@aims4:~$ python -c 'import acme_diags; print(acme_diags.__version__)'
v1.2.1
```

**Comment by kaabre** [comment]
@zshaheen Hmm no despite running the conda update/upgrade acme_diags command it's still v1.2.0.  Is there something else I need to do?  I am in the non-dev environment.

**Comment by zshaheen** [comment]
Try the following two commands. Below is also the output:
* `conda remove acme_diags --force`: `--force` only removes`acme_diags`, but keeps the dependencies.
* `conda install acme_diags -c conda-forge -c uvcdat -c acme`: Installs the latest `acme_diags` by looking though the `conda-forge`, `uvcdat`, and `acme` channel in that order.

Output:
```
(acme_diags_env_old) shaheen2@aims4:~$ conda remove acme_diags --force
Fetching package metadata ...................

Package plan for package removal in environment /export_2/shaheen2/anaconda2/envs/acme_diags_env_old:

The following packages will be REMOVED:

    acme_diags: 1.2.1-0 acme

Proceed ([y]/n)? y

(acme_diags_env_old) shaheen2@aims4:~$ conda install acme_diags -c conda-forge -c uvcdat -c acme
Fetching package metadata ...................
Solving package specifications: .

Package plan for installation in environment /export_2/shaheen2/anaconda2/envs/acme_diags_env_old:

The following NEW packages will be INSTALLED:

    acme_diags: 1.2.1-0 acme

Proceed ([y]/n)? y

(acme_diags_env_old) shaheen2@aims4:~$ python -c 'import acme_diags; print(acme_diags.__version__)'
v1.2.1
```

**Comment by kaabre** [comment]
It still only wants to install the 1.2.0 version for some reason?

(acme_diags_env) leebrent@edison11:~> conda remove acme_diags --force         
Solving environment: done


==> WARNING: A newer version of conda exists. <==
  current version: 4.4.7
  latest version: 4.4.10

Please update conda by running

    $ conda update -n base conda



## Package Plan ##

  environment location: /global/homes/l/leebrent/.conda/envs/acme_diags_env

  removed specs: 
    - acme_diags


The following packages will be REMOVED:

    acme_diags: 1.2.0-0 acme

Proceed ([y]/n)? y

Preparing transaction: done
Verifying transaction: done
Executing transaction: done
(acme_diags_env) leebrent@edison11:~> conda install acme_diags -c conda-forge -c uvcdat -c acme
Solving environment: done


==> WARNING: A newer version of conda exists. <==
  current version: 4.4.7
  latest version: 4.4.10

Please update conda by running

    $ conda update -n base conda



## Package Plan ##

  environment location: /global/homes/l/leebrent/.conda/envs/acme_diags_env

  added / updated specs: 
    - acme_diags


**The following NEW packages will be INSTALLED:

    acme_diags: 1.2.0-0                    acme**

**Comment by zshaheen** [comment]
Well, I guess the only thing to do is to delete your env and start again.
1. `source deactivate acme_diags_env`
2. `conda remove -n acme_diags_env --all`
3. Follow the steps under 'Latest stable release' [here](https://acme-climate.github.io/acme_diags/docs/html/install-config-run.html#installation).

That 'should' actually do it.

**Comment by kaabre** [comment]
All right, back to the beginning -- and it did!  Version is now 1.2.1, I've submitted a test to make sure it works as expected.

**Comment by zshaheen** [comment]
@kaabre Nice, this is good to know!

**Comment by chengzhuzhang** [comment]
@kaabre I noticed that there are no html for tables generated at: https://portal.nersc.gov/project/PNNL-PJR/bren937/scratch/TEST/newvers/
However, when I try a model vs model run myself. Tables are actually generated...Would you let me know if you still missing the tables with v1.2.1 runs? Thanks!

**Comment by kaabre** [comment]
Hi Jill,

This is linked to issue #123 that I already noted – changing the name of the directories affects the path of the html tables.  They still exist, but the links to them through the viewer are broken.  Also, this output was run before the DISPLAY variable issue was fixed in version 1.2.1

The tables for the nomachine test are located here: https://portal.nersc.gov/project/PNNL-PJR/bren937/scratch/TEST/newvers/nomachinetest/viewer/table-data/ANN_metrics_table.html – basically you have to change the path manually to see them from the viewer links.

**Comment by chengzhuzhang** [comment]
Thank you, Lee! @kaabre We will work on the #123 and hopefully fix the issue all together...

**Comment by kaabre** [comment]
Appreciated!

**Comment by zshaheen** [comment]
@chengzhuzhang @kaabre Just letting you both know, there's a fix for #123 here: 587e5af0521493c4264fc0babb927f150d60d980. I just need to merge that branch into master soon. I'll let you both know when that's done.

@kaabre We plan on another release sometime in early to mid March, and that will have the fix for #123.

**Comment by chengzhuzhang** [comment]
Pull request #139 resolves this issue
